base/Base.jl

```diff
@@ -35,6 +35,7 @@ getproperty(x::Tuple, f::Int) = (@inline; getfield(x, f))
 setproperty!(x::Tuple, f::Int, v) = setfield!(x, f, v) # to get a decent error
 
 getproperty(x, f::Symbol) = (@inline; getfield(x, f))
+getproperty(f::Union{Symbol,String,Int}) = Base.Fix2(getproperty, f)
```

Comment:
```suggestion
getproperty(f::Symbol) = x -> getproperty(x, f)
```

In a case when the type of field name is known statically but the field value is determined at runtime, the anonymous version will generate the (almost) same amount of code as the `Fix2` version. But the anonymous function is more efficient when the name is known as compile-time constant because it can propagate constant information.